test/java/seedu/addressbook/commands/DeleteCommandTest.java

```diff
@@ -0,0 +1,169 @@
+                new UniqueTagList());
+
+        emptyAddressBook = new AddressBook();
+
+        addressBook = new AddressBook();
+        addressBook.addPerson(johnDoe);
+        addressBook.addPerson(janeDoe);
+        addressBook.addPerson(davidGrant);
+
+        emptyDisplayList = new ArrayList<ReadOnlyPerson>();
+
+        listWithEveryone = new ArrayList<ReadOnlyPerson>();
+        listWithEveryone.add(johnDoe);
+        listWithEveryone.add(janeDoe);
+        listWithEveryone.add(davidGrant);
+
+        listWithSurnameDoe = new ArrayList<ReadOnlyPerson>();
+        listWithSurnameDoe.add(johnDoe);
+        listWithSurnameDoe.add(janeDoe);
+    }
+
+    @Test
+    public void execute_emptyAddressBook_returnsPersonNotFoundMessage() {
+        DeleteCommand command = createDeleteCommand(1, emptyAddressBook, listWithEveryone);
+
+        assertCommandBehaviour(command, Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK, emptyAddressBook,
+                emptyAddressBook);
+    }
+
+    @Test
+    public void execute_noPersonDisplayed_returnsInvalidIndexMessage() {
+        DeleteCommand command = createDeleteCommand(1, addressBook, emptyDisplayList);
+
+        assertCommandBehaviour(command, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX, addressBook,
+                addressBook);
+    }
+
+    @Test
+    public void execute_targetPersonNotInAddressBook_returnsPersonNotFoundMessage()
+            throws IllegalValueException {
+        Person notInAddressBookPerson = new Person(new Name("Alien"), new Phone("10000", false),
+                new Email("a@b.c", false), new Address("Core of the Sun", false), new UniqueTagList());
+        List<ReadOnlyPerson> listWithPersonNotInAddressBook = Arrays.asList(notInAddressBookPerson);
+
+        DeleteCommand command = createDeleteCommand(1, addressBook, listWithPersonNotInAddressBook);
+
+        assertCommandBehaviour(command, Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK, addressBook, addressBook);
+    }
+
+    @Test
+    public void execute_indexZero_returnsInvalidIndexMessage() {
+        DeleteCommand command = createDeleteCommand(0, addressBook, listWithEveryone);
+
+        assertCommandBehaviour(command, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX, addressBook,
+                addressBook);
+    }
+
+    @Test
+    public void execute_indexNegative_returnsInvalidIndexMessage() {
+        DeleteCommand command = createDeleteCommand(-1, addressBook, listWithEveryone);
+
+        assertCommandBehaviour(command, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX, addressBook,
+                addressBook);
+    }
+
+    @Test
+    public void execute_indexMoreThanListSize_returnsInvalidIndexMessage() {
+        DeleteCommand command = createDeleteCommand(listWithSurnameDoe.size() + 1, addressBook,
+                listWithSurnameDoe);
+
+        assertCommandBehaviour(command, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX, addressBook,
+                addressBook);
+    }
```

Comment:
See if you can put these three 'invalid index' cases in one method. All three has similar code, so possibly extract into a utility method?